hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/MetadataStore.java

```diff
@@ -63,14 +63,19 @@
    * Deletes exactly one path, leaving a tombstone to prevent lingering,
    * inconsistent copies of it from being listed.
    *
+   * Deleting an entry with a tombstone needs a
+   * {@link org.apache.hadoop.fs.s3a.s3guard.S3Guard.TtlTimeProvider} because
+   * the lastUpdated field of the record has to be updated to <pre>now</pre>.
+   *
    * @param path the path to delete
+   * @param ttlTimeProvider
    * @throws IOException if there is an error
    */
-  void delete(Path path) throws IOException;
+  void delete(Path path, ITtlTimeProvider ttlTimeProvider) throws IOException;
```

Comment:
Declare this as @Nullable (assuming it is), cover in javadocs.

Reply:
Of course, if deleting with tombstones means it is not nullable, then only the javadocs are needed: we just need to make sure a null value is ever passed in

Reply:
Iff you delete with tombstones it should not be not null.

Reply:
But it is better if it's not nullable, but the implementations have to be sure that this value is not null.